.github/workflows/docs-check.yml

```diff
@@ -62,21 +73,23 @@ jobs:
 
       - name: Install dependencies
         run: |
-          pip install -e . -U -q -r requirements/docs.txt -f https://download.pytorch.org/whl/torch_stable.html
           # install Texlive, see https://linuxconfig.org/how-to-install-latex-on-ubuntu-20-04-focal-fossa-linux
           sudo apt-get update
           sudo apt-get install -y texlive-latex-extra dvipng texlive-pictures texlive-fonts-recommended cm-super
-          python --version
           pip --version
+          pip install -e . -U -q -r requirements/docs.txt --find-links $PYTORCH_URL -f $PYPI_CACHE
           pip list
         shell: bash
 
       - name: Make Documentation
-        # First run the same pipeline as Read-The-Docs
         working-directory: ./docs
-        run: |
-          make clean
-          make html --debug SPHINXOPTS="-W --keep-going" -b linkcheck
+        run: make html --debug SPHINXOPTS="-W --keep-going"
+
+      - name: Check External Links (Optional)
+        working-directory: ./docs
+        run: make --jobs $(nproc) linkcheck
+        # ToDO: comment on PR if any link failed
```

Comment:
this would be great to have as action in our utils as we switch to this split build in several other projects